dotcom-rendering/src/web/components/YoutubeBlockComponent.importable.tsx

```diff
@@ -94,22 +91,16 @@ export const YoutubeBlockComponent = ({
 	);
 
 	useEffect(() => {
-		import(
-			/* webpackChunkName: "cmp" */ '@guardian/consent-management-platform'
-		)
-			.then(
-				(module: { onConsentChange: (callback: Callback) => void }) => {
-					module.onConsentChange((newConsent: ConsentState) => {
-						setConsentState(newConsent);
-					});
-				},
-			)
-			.catch((error) => {
-				window.guardian.modules.sentry.reportError(
-					new Error(`Error: ${error}`),
-					'youtube-consent',
-				);
+		try {
+			window.guCmpHotFix.onConsentChange((newConsent: ConsentState) => {
```

Comment:
@sndrs @ashishpuliyel Where are you at on this pattern? We need some way to introduce cmp code inside a file which gets rendered on the server

Reply:
this would be one for @kenoir – we discussed the need, not sure of the priority though. 

it shouldn't be hard to implement though if you want to take a crack @DavidLawes? i think a similar check for `window` around [here](https://github.com/guardian/consent-management-platform/blob/3ca4f9113495f9268ac34b41338850be5604a8af/src/index.ts#L32) might be enough...

Reply:
@oliverlloyd is [this](https://github.com/guardian/consent-management-platform/pull/555) the kind of thing you were thinking of?

Reply:
just to update this – @DavidLawes and i spoke offline, he's going to look into why the import inside `useEffect` failed before (seems like it should work!) and @kenoir is also going to push forward with https://github.com/guardian/consent-management-platform/pull/555 because it should really not fail if it's imported in server-side code anyway

Reply:
To update here: I've changed to a dynamic import and now both server and client side rendering appears to work ok. Cmp is dynamically loaded on the client, consent is set correctly and I can play videos.

My only concern / confusion is about the number of times cmp is requested:

page with a single video (cmp requested 3 times):
https://www.theguardian.com/uk-news/2020/jan/09/prince-harry-meghan-decision
![Screenshot 2022-02-24 at 12 21 10](https://user-images.githubusercontent.com/45561419/155522983-1c23844f-ffc9-49aa-a95e-310aa012e214.png)

page with multiple videos (cmp requested once):
![Screenshot 2022-02-24 at 12 21 31](https://user-images.githubusercontent.com/45561419/155523035-f31865c4-f0c0-456f-95b2-12332bbb618c.png)


@oliverlloyd suggested this could be a result of being in-between the App.tsx --> Islands migration. 

If the way I've dynamically imported cmp looks ok, maybe it's worth pushing this solution and investigating the multiple imports as a follow-on task? cc @sndrs 

Also @kenoir - I'd attempted to make cmp server safe, your implementation looks much neater & well-informed than mine :)

Reply:
^^ re cmp loading, I think there's something else going on, on `main` this [page](https://www.theguardian.com/uk-news/2020/jan/09/prince-harry-meghan-decision) already loads 2 cmp bundles. I think cmp bundle loading should be investigated separately from this change...